src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -794,6 +797,81 @@ public void testJavaParseExceptionSeverityDefaultError() throws Exception {
         verify(checker, files, expectedViolation);
     }
 
+    @Test
+    public void testThenComparingWithMethodReference() {
+        final List<TestCheck> checks = Arrays.asList(
+                new TestCheck("check2"),
+                new TestCheck("check1"),
+                new TestCheck("check3")
+        );
+
+        checks.sort(Comparator
+                .comparing(TestCheck::getPriority)
+                .thenComparing(TestCheck::getCustomIdWrapper));
+
+        Assertions.assertEquals("check1", checks.get(0).getId(),
+                "First element should be 'check1'");
+        Assertions.assertEquals("check2", checks.get(1).getId(),
+                "Second element should be 'check2'");
+        Assertions.assertEquals("check3", checks.get(2).getId(),
+                "Third element should be 'check3'");
+    }
+
+    public static class TestCheck extends AbstractCheck {
```

Comment:
Please kill mutation by existing Check.
If not possible, we will need long explanation over this class on why it is required.

Reply:
violations might come to result report in special order duet to this sorting, try to make such config in Input file where missed sorting by ID makes order or violation diffrent, so you can catch it test assert.